kernel/syscall.c

```diff
@@ -2933,6 +2933,30 @@ long myst_syscall_prlimit64(
             old_rlim->rlim_max = MYST_PROCESS_MAX_STACK_SIZE;
         }
     }
+    else if (resource == RLIMIT_NPROC)
+    {
+        if (old_rlim)
+        {
+            old_rlim->rlim_cur = __myst_kernel_args.max_threads;
+            old_rlim->rlim_max = __myst_kernel_args.max_threads;
+        }
+    }
+    else if (resource == RLIMIT_AS)
+    {
+        if (old_rlim)
+        {
+            old_rlim->rlim_cur = RLIM_INFINITY;
+            old_rlim->rlim_max = RLIM_INFINITY;
+        }
+    }
+    else if (resource == RLIMIT_FSIZE)
+    {
+        if (old_rlim)
+        {
+            old_rlim->rlim_cur = RLIM_INFINITY;
+            old_rlim->rlim_max = RLIM_INFINITY;
+        }
+    }
```

Comment:
RLIMIT_DATA is relevant too. The app won't be able to allocate more than the Mystikos mman region for sure. I don't think it's a must-have for the implementation to enforce the memory allocation limit, but providing a "hint" on how much memory is available might allow some apps or runtimes to adjust their memory usage accordingly. Maybe the implementation can report a percentage of the mman region size, say 70%?